go.mod

```diff
@@ -24,13 +24,15 @@ require (
 	golang.org/x/oauth2 v0.20.0
 	k8s.io/api v0.29.4
 	k8s.io/apimachinery v0.29.4
-	k8s.io/client-go v0.29.4
+	k8s.io/client-go v1.5.2
 	k8s.io/klog/v2 v2.120.1
-	k8s.io/utils v0.0.0-20240102154912-e7106e64919e
-	knative.dev/pkg v0.0.0-20240219120257-9227ebb57a4e
+	k8s.io/utils v0.0.0-20240310230437-4693a0247e57
+	knative.dev/pkg v0.0.0-20240404013351-5d4af76051e4
 	sigs.k8s.io/controller-runtime v0.17.3
 )
 
+replace k8s.io/client-go => k8s.io/client-go v0.29.4
```

Comment:
Add this because below message are reported when  running `go mod tidy`
```
go: github.com/konflux-ci/integration-service/cmd imports
	k8s.io/client-go/kubernetes/scheme: package k8s.io/client-go/kubernetes/scheme provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/cmd imports
	k8s.io/client-go/plugin/pkg/client/auth: package k8s.io/client-go/plugin/pkg/client/auth provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/internal/controller/buildpipeline imports
	k8s.io/client-go/util/retry: package k8s.io/client-go/util/retry provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/api/v1beta2 tested by
	github.com/konflux-ci/integration-service/api/v1beta2.test imports
	k8s.io/client-go/rest: package k8s.io/client-go/rest provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/cache imports
	sigs.k8s.io/controller-runtime/pkg/client imports
	k8s.io/client-go/metadata: package k8s.io/client-go/metadata provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/cmd imports
	sigs.k8s.io/controller-runtime/pkg/metrics/server imports
	k8s.io/client-go/util/cert: package k8s.io/client-go/util/cert provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/cmd/snapshotgc imports
	sigs.k8s.io/controller-runtime/pkg/client/config imports
	k8s.io/client-go/tools/clientcmd: package k8s.io/client-go/tools/clientcmd provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/cmd/snapshotgc imports
	sigs.k8s.io/controller-runtime/pkg/client/config imports
	k8s.io/client-go/tools/clientcmd/api: package k8s.io/client-go/tools/clientcmd/api provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/internal/controller imports
	sigs.k8s.io/controller-runtime/pkg/manager imports
	k8s.io/client-go/tools/leaderelection: package k8s.io/client-go/tools/leaderelection provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/internal/controller imports
	sigs.k8s.io/controller-runtime/pkg/manager imports
	k8s.io/client-go/tools/leaderelection/resourcelock: package k8s.io/client-go/tools/leaderelection/resourcelock provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/internal/controller imports
	sigs.k8s.io/controller-runtime/pkg/manager imports
	k8s.io/client-go/tools/record: package k8s.io/client-go/tools/record provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/metrics imports
	sigs.k8s.io/controller-runtime/pkg/metrics imports
	k8s.io/client-go/tools/metrics: package k8s.io/client-go/tools/metrics provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/metrics imports
	sigs.k8s.io/controller-runtime/pkg/metrics imports
	k8s.io/client-go/util/workqueue: package k8s.io/client-go/util/workqueue provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/cache imports
	sigs.k8s.io/controller-runtime/pkg/client imports
	sigs.k8s.io/controller-runtime/pkg/client/apiutil imports
	k8s.io/client-go/discovery: package k8s.io/client-go/discovery provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/cache imports
	sigs.k8s.io/controller-runtime/pkg/client imports
	sigs.k8s.io/controller-runtime/pkg/client/apiutil imports
	k8s.io/client-go/dynamic: package k8s.io/client-go/dynamic provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/cache imports
	sigs.k8s.io/controller-runtime/pkg/client imports
	sigs.k8s.io/controller-runtime/pkg/client/apiutil imports
	k8s.io/client-go/restmapper: package k8s.io/client-go/restmapper provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/cache imports
	sigs.k8s.io/controller-runtime/pkg/client tested by
	sigs.k8s.io/controller-runtime/pkg/client.test imports
	k8s.io/client-go/applyconfigurations/core/v1: package k8s.io/client-go/applyconfigurations/core/v1 provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/cache imports
	sigs.k8s.io/controller-runtime/pkg/client tested by
	sigs.k8s.io/controller-runtime/pkg/client.test imports
	k8s.io/client-go/kubernetes: package k8s.io/client-go/kubernetes provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/cmd imports
	github.com/tektoncd/pipeline/pkg/apis/pipeline/v1 imports
	knative.dev/pkg/kmeta imports
	k8s.io/client-go/tools/cache: package k8s.io/client-go/tools/cache provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/cmd/snapshotgc tested by
	github.com/konflux-ci/integration-service/cmd/snapshotgc.test imports
	sigs.k8s.io/controller-runtime/pkg/client/fake imports
	k8s.io/client-go/testing: package k8s.io/client-go/testing provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/internal/controller imports
	sigs.k8s.io/controller-runtime/pkg/manager imports
	sigs.k8s.io/controller-runtime/pkg/internal/recorder imports
	k8s.io/client-go/kubernetes/typed/core/v1: package k8s.io/client-go/kubernetes/typed/core/v1 provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/internal/controller imports
	sigs.k8s.io/controller-runtime/pkg/manager imports
	sigs.k8s.io/controller-runtime/pkg/leaderelection imports
	k8s.io/client-go/kubernetes/typed/coordination/v1: package k8s.io/client-go/kubernetes/typed/coordination/v1 provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/api/v1beta2 tested by
	github.com/konflux-ci/integration-service/api/v1beta2.test imports
	sigs.k8s.io/controller-runtime/pkg/envtest imports
	k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset imports
	k8s.io/client-go/util/flowcontrol: package k8s.io/client-go/util/flowcontrol provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/cache imports
	sigs.k8s.io/controller-runtime/pkg/client imports
	sigs.k8s.io/controller-runtime/pkg/client/apiutil tested by
	sigs.k8s.io/controller-runtime/pkg/client/apiutil.test imports
	k8s.io/client-go/kubernetes/fake: package k8s.io/client-go/kubernetes/fake provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/internal/controller imports
	sigs.k8s.io/controller-runtime/pkg/manager imports
	sigs.k8s.io/controller-runtime/pkg/internal/recorder tested by
	sigs.k8s.io/controller-runtime/pkg/internal/recorder.test imports
	k8s.io/client-go/tools/reference: package k8s.io/client-go/tools/reference provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/api/v1alpha1 imports
	sigs.k8s.io/controller-runtime imports
	sigs.k8s.io/controller-runtime/pkg/builder imports
	sigs.k8s.io/controller-runtime/pkg/source tested by
	sigs.k8s.io/controller-runtime/pkg/source.test imports
	k8s.io/client-go/informers: package k8s.io/client-go/informers provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/cmd imports
	github.com/openshift-pipelines/pipelines-as-code/pkg/apis/pipelinesascode/v1alpha1 imports
	knative.dev/pkg/apis/duck/v1 imports
	knative.dev/pkg/apis/duck tested by
	knative.dev/pkg/apis/duck.test imports
	k8s.io/client-go/dynamic/fake: package k8s.io/client-go/dynamic/fake provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
go: github.com/konflux-ci/integration-service/api/v1beta2 tested by
	github.com/konflux-ci/integration-service/api/v1beta2.test imports
	sigs.k8s.io/controller-runtime/pkg/envtest imports
	k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset imports
	k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1 imports
	k8s.io/apiextensions-apiserver/pkg/client/applyconfiguration/apiextensions/v1 imports
	k8s.io/client-go/applyconfigurations/meta/v1: package k8s.io/client-go/applyconfigurations/meta/v1 provided by k8s.io/client-go at latest version v0.30.1 but not at required version v1.5.2
```
So I add `replace` as used in https://github.com/openshift-pipelines/pipelines-as-code/blob/main/go.mod#L58